include/rpm/rpmts.h

```diff
@@ -359,7 +360,18 @@ rpmRC rpmtxnImportPubkey(rpmtxn txn, const unsigned char * pkt, size_t pktlen);
  * 			RPMRC_NOKEY on invalid keyid
  * 			RPMRC_FAIL on other failure
  */
-rpmRC rpmtxnDeletePubkey(rpmtxn txn, const char *keyid);
+rpmRC rpmtxnDeletePubkeyByID(rpmtxn txn, const char *keyid);
+
+/** \ingroup rpmts
+ * Delete public key from transaction keystore.
+ * @param txn           transaction handle
+ * @param key           public key
+ * @return              RPMRC_OK on success
+ * 			RPMRC_NOTFOUND if key not found
+ * 			RPMRC_NOKEY on invalid keyid
```

Comment:
You can drop the NOKEY return, if it takes a pubkey then we already know it's a key.
Getting rid of that ugly kludge NOKEY == INVALID mapping is actually another reason to just drop the ByID variant.